### ISSUE TITLE
Build pack is installing the wrong version of Ruby.

### ISSUE THREAD

**Issue by dmikusa-pivotal** [cloudfoundry/ruby-buildpack#63]
>What version of Cloud Foundry are you using?

CF v235.5.50

>What version of the buildpack you are using?

1.6.16 & 1.6.20

>If you were attempting to accomplish a task, what was it you were attempting to do?

Deploy an application that works on 1.6.16 and fails on 1.6.20.

>What did you expect to happen?

Build pack to pick the version set in Gemfile-APP

>What was the actual behavior?

Build pack seems to pick the default value.

Attaching a test case.  Steps to reproduce:

1.) Target PWS or something with Ruby Build pack 1.6.20 or 21 deployed.
2.) Run `cf push`
3.) You should get a failure message like this.

```
Downloading ruby_buildpack...
Downloaded ruby_buildpack
Creating container
Successfully created container
Downloading app package...
Staging...
Downloaded app package (1.7K)
-------> Buildpack version 1.6.21
       Downloaded [file:///tmp/buildpacks/52950f43493e01421589bf265ae8c07e/dependencies/https___buildpacks.cloudfoundry.org_concourse-binaries_bundler_bundler-1.12.5.tgz]
-----> Compiling Ruby/Rack
       Downloaded [file:///tmp/buildpacks/52950f43493e01421589bf265ae8c07e/dependencies/https___buildpacks.cloudfoundry.org_concourse-binaries_ruby_ruby-2.3.1-linux-x64.tgz]
-----> Using Ruby version: ruby-2.3.1
-----> Installing dependencies using bundler 1.12.5
       Downloaded [file:///tmp/buildpacks/52950f43493e01421589bf265ae8c07e/dependencies/https___buildpacks.cloudfoundry.org_ruby_binaries_cflinuxfs2_libyaml-0.1.6.tgz]
       Running: bundle install --without development:test --path vendor/bundle --binstubs vendor/bundle/bin -j4 --deployment
       Your Ruby version is 2.3.1, but your Gemfile specified 2.2.4
       Bundler Output: Your Ruby version is 2.3.1, but your Gemfile specified 2.2.4
 !
 !     Failed to install gems via Bundler.
 !
Failed to compile droplet
Exit status 223
Staging failed: Exited with status 223
Destroying container
Successfully destroyed container
```

Note: `Your Ruby version is 2.3.1, but your Gemfile specified 2.2.4`

Please confirm where necessary:
* [X] I have included a log output 
* [X] My log includes an error message 
* [X] I have included steps for reproduction

**Comment by jchesterpivotal** [comment]
@dmikusa-pivotal we can't see the attached test case.

**Comment by dmikusa-pivotal** [comment]
Sorry, trying again.

[ruby-sample.tar.gz](https://github.com/cloudfoundry/ruby-buildpack/files/435344/ruby-sample.tar.gz)

**Comment by sesmith177** [comment]
This has been resolved with 4801e61ecc19697e92aff86301b32a4d98f9f1b8 in develop. It will be released in 1.6.22, [which has been scheduled](https://www.pivotaltracker.com/story/show/128993999).
